File: Python/128LongestConsecutiveSequence/Solution.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num_set = set(nums) 
        longest_seq = 0
        for num in num_set:
            if num - 1 not in num_set:
                current_num = num
                current_streak = 1
            while current_num + 1 in num_set:
                current_num += 1
                current_streak += 1
            longest_seq = max(longest_seq, current_streak)
        return longest_seq

File: Java/128LongestConsecutiveSequence/Solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int longest_seq = 0;
        if (nums.length == 0) {
            longest_seq = 0; 
        }
        Arrays.sort(nums); 
        int longest = 1; 
        int streak = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i - 1]) { 
                if (nums[i] - nums[i - 1] == 1) {
                    streak++;
                } else {
                    longest = Math.max(longest, streak); 
                    streak = 1; 
                }
            }
        }
        if (nums.length > 0) {
            longest_seq = Math.max(longest, streak);
        }
        return longest_seq;
    }
}
